src/async_graphql_hyper.rs

```diff
@@ -126,27 +126,30 @@ impl GraphQLResponse {
     Ok(response)
   }
 
-  /// Sets the `cache_control` max_age for a given `GraphQLResponse`.
+  /// Sets the `cache_control` for a given `GraphQLResponse`.
   ///
   /// The function modifies the `GraphQLResponse` to set the `cache_control` `max_age`
-  /// to the specified `min_cache` value.
+  /// to the specified `min_cache` value and `public` flag to `!cache_private`
   ///
   /// # Arguments
   ///
   /// * `res` - The GraphQL response whose `cache_control` is to be set.
   /// * `min_cache` - The `max_age` value to be set for `cache_control`.
+  /// * `cache_private` - The negation of `public` flag to be set for `cache_control`.
   ///
   /// # Returns
   ///
-  /// * A modified `GraphQLResponse` with updated `cache_control` `max_age`.
-  pub fn set_cache_control(mut self, min_cache: i32) -> GraphQLResponse {
+  /// * A modified `GraphQLResponse` with updated `cache_control` `max_age` and `public` flag.
+  pub fn set_cache_control(mut self, min_cache: i32, cache_private: bool) -> GraphQLResponse {
```

Comment:
We could use `cache_public` here since we're doing `!cache_private` where the variable is used.

It'd make the core more concise and readable.

Reply:
Yes, I understand. Updated it.